src/riak_kv_pb_timeseries.erl

```diff
@@ -255,6 +261,112 @@ wait_until_active(Table, State, Seconds) ->
 %% functions called from check_table_and_call, one per ts* request
 %% ---------------------------------------------------
 
+
+%%
+%% INSERT statements, called from check_table_and_call.
+%%
+-spec make_insert_response(module(), #riak_sql_insert_v1{}) ->
+                           #tsqueryresp{} | #rpberrorresp{}.
+make_insert_response(Mod, #riak_sql_insert_v1{'INSERT' = Table, fields = Fields, values = Values}) ->
+    case lookup_field_positions(Mod, Fields) of
+    {ok, Positions} ->
+        Empty = make_empty_row(Mod),
+        case xlate_insert_to_putdata(Values, Positions, Empty) of
+            {error, ValueReason} ->
+                make_rpberrresp(?E_BAD_QUERY, ValueReason);
+            {ok, Data} ->
+                insert_putreqs(Mod, Table, Data)
+	    end;
+    {error, FieldReason} ->
+        make_rpberrresp(?E_BAD_QUERY, FieldReason)
+    end.
+
+insert_putreqs(Mod, Table, Data) ->
+    case catch validate_rows(Mod, Data) of
+        [] ->
+            case put_data(Data, Table, Mod) of
+                0 ->
+                    #tsqueryresp{};
+                ErrorCount ->
+                    failed_put_response(ErrorCount)
+            end;
+        BadRowIdxs when is_list(BadRowIdxs) ->
+            validate_rows_error_response(BadRowIdxs)
+    end.
+
+%%
+%% Return an all-null empty row ready to be populated by the values
+%%
+-spec make_empty_row(module()) -> tuple(undefined).
+make_empty_row(Mod) ->
+    Positions = Mod:get_field_positions(),
+    list_to_tuple(lists:duplicate(length(Positions), undefined)).
+
+%%
+%% Lookup the index of the field names selected to insert.
+%%
+%% This *requires* that once schema changes take place the DDL fields are left in order.
+%%
+-spec lookup_field_positions(module(), [field_identifier()]) ->
+                           {ok, [pos_integer()]} | {error, string()}.
+lookup_field_positions(Mod, FieldIdentifiers) ->
+    case lists:foldl(
+	   fun({identifier, FieldName}, {Good, Bad}) ->
+		   case Mod:is_field_valid(FieldName) of
+               false ->
+                   {Good, [FieldName | Bad]};
+               true ->
+                   {[Mod:get_field_position(FieldName) | Good], Bad}
+		   end
+	   end, {[], []}, FieldIdentifiers)
+    of
+        {Positions, []} ->
```

Comment:
This requires an iteration of `Positions`, this is probably a small list but `Positions /= []` doesn't require the iteration.